app-backend/database/src/main/scala/com/azavea/rf/database/tables/Buckets.scala

```diff
@@ -239,4 +282,40 @@ object Buckets extends TableQuery(tag => new Buckets(tag)) with LazyLogging {
       sceneBucketJoinQuery.delete
     }
   }
+
+  /** Removes scenes from bucket
+    *
+    * @param sceneIds Seq[UUID] primary key of scenes to remove from bucket
+    * @param bucketId UUID primary key of bucket these scenes will be removed from
+    */
+  def deleteScenesFromBucket(sceneIds: Seq[UUID], bucketId: UUID)
+    (implicit database: DB): Future[Int] = {
+
+    val sceneBucketJoinQuery = for {
+      s <- ScenesToBuckets if s.sceneId.inSet(sceneIds) && s.bucketId === bucketId
+    } yield s
+
+    database.db.run {
+      sceneBucketJoinQuery.delete
+    }
+  }
+
+  /** Removes all scenes from bucket, then adds all specified scenes to it
+    *
+    * @param sceneIds Seq[UUID] primary key of scenes to add to bucket
+    * @param bucketId UUID primary key of bucket to remove all scenes from and add specified scenes to
+    * @return Scenes that were added
+    */
+  def replaceScenesInBucket(sceneIds: Seq[UUID], bucketId: UUID)
+    (implicit database: DB): Future[Iterable[Scene.WithRelated]] = {
+
+    database.db.run {
+      (for {
+        _ <- ScenesToBuckets.filter(sb => sb.bucketId === bucketId).delete
+        _ <- ScenesToBuckets ++= sceneIds.map(sceneId => SceneToBucket(sceneId, bucketId))
+      } yield()).transactionally
```

Comment:
Tangential question: should we be using `+=` or `forceInsert`?

I will switch to this pattern, which I like more because we're using DataModel objects and not inserting directly in to the database.